lib/Wikia/src/Domain/User/Preferences/UserPreferences.php

```diff
@@ -80,4 +83,28 @@ public function getGlobalPreferences() {
 	public function getLocalPreferences() {
 		return $this->localPreferences;
 	}
+
+	/**
+	 * This toggle is to flag the object as being not suitable for saving.
+	 *
+	 * @param boolean state true to set the preferences to read only
+	 */
+	public function setReadOnly( $state ) {
+		$this->readOnly = $state;
+		return $this;
+	}
+
+	/**
+	 * @return boolean the read only state
+	 */
+	public function getReadOnly() {
```

Comment:
Do we need this method? For accessing a boolean I think `isReadOnly` is more intuitive.